### DISCUSSION TITLE
Help on “use client” properties on a redirected page

### DISCUSSION THREAD

**Discussion by sandervaneck** [vercel/next.js#74035] [Help]
I have created a nextjs app in react with an integration with stripjs for receiving payments.

On finishing the payment, the app redirects to a page, which should have “use client” usability, so I can call useState().

Please help out.

**Comment by icyJoseph** [comment]
What problem are you having? You would just need to have `use client` on your page...

**Reply by sandervaneck** [reply]
So: I have a payment button (which has an integration with StripeJS).

It's successurl = localhost:3000/api/results/[clientID]

Now, when rendering this page, when I use "use server" I can't use useState()
When I use "use client" I can't use the clientID from the url.

**Reply by adixoo** [reply]
You need separate components,
Simply create two components first one is server component and second one is client component, now import client component inside server component and pass the Client id from server component to client component as props
Now you can use useState inside client component

**Reply by icyJoseph** [reply]
> Now, when rendering this page, when I use "use server" I can't use useState()

`use server` is a directive to create server actions, not to create server components.

`use client` is used to bring interactivity, you have that part right.

So, as the other commenter says, you can break down your components, and gain the ability to do Server Components and Client Components. 

I guess, you do have a server action anyway, couldn't you invoke it within a client component, and then `.then` or `await` the server action, and if successful setState?
